test/pkg/environment/environment.go

```diff
@@ -25,7 +26,7 @@ import (
 	"github.com/aws/karpenter/pkg/utils/project"
 )
 
-var clusterNameFlag = *flag.String("cluster-name", env.WithDefaultString("CLUSTER_NAME", ""), "Cluster name enables discovery of the testing environment")
+var clusterNameFlag = flag.String("cluster-name", env.WithDefaultString("CLUSTER_NAME", ""), "Cluster name enables discovery of the testing environment")
```

Comment:
Why is this using flag? It looks like this will only ever get the env variable since you never call flag.Parse. Might as well just use:

```suggestion
var clusterName = env.WithDefaultString("CLUSTER_NAME", "")
```
or directly in `DiscoverClusterName()`

Reply:
The test library calls flags.Parse() for us =/. I can't call it again without creating problem. I like your solution though.